sql/src/main/java/io/crate/analyze/AlterTableParameterInfo.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze;
+
+import com.google.common.collect.ImmutableMap;
+import org.elasticsearch.common.settings.Setting;
+
+import javax.annotation.concurrent.Immutable;
+import javax.annotation.concurrent.ThreadSafe;
+
+@Immutable
+@ThreadSafe
+public class AlterTableParameterInfo extends TableParameterInfo {
```

Comment:
Would it be possible to re-use the `TableParameterInfo` by making the `supportedSettings` an argument to the constructor, so that we simply have 2 different instances?

Reply:
ok @mfussenegger, pushed some changes on that direction , not sure I prefer it to be honest.
Anyway, have a look and we discuss.

@andreidan @seut what's your opinion on this? (check both commits)

Reply:
fyi: refactored in another issue #7911